src/preamble.js

```diff
@@ -1266,4 +1266,12 @@ function runMemoryInitializer() {
 var asyncifyStubs = {};
 #endif
 
+#if PTHREADS
+// Listen to errors from all threads and report it
+self.addEventListener("error", (e) => {
```

Comment:
This errors out with `self is not defined`, but somehow this works on my machine. What should I use instead? `window` doesn't work...

Reply:
Because you are using `postMessage` to send the events over to the main thread I don't think this should be needed.  Won't the events show up in the `worker.onmessage` handler in `library_pthread.js`?  That is where messages from workers get handled IIUC.

Reply:
Sorry for the delayed response, and thanks for the review.

I deleted this and added the same code in `worker. onmessage` or `worker.onerror` in `library_pthread.js` but I don't know how to make it work. Is this the right place you're talking about? https://github.com/emscripten-core/emscripten/blob/518d9fea335f7cd2b5771e43df76e0535c6df5dd/src/library_pthread.js#L534-L539

What I added:
```diff
diff --git a/src/library_pthread.js b/src/library_pthread.js
index 170d9f5f0..b61305199 100644
--- a/src/library_pthread.js
+++ b/src/library_pthread.js
@@ -532,11 +532,17 @@ var LibraryPThread = {
     // out its message handler here.  This avoids having to check in each of
     // the onmessage handlers if the message was coming from valid worker.
     worker.onmessage = (e) => {
+      console.log('terminateWorker: worker.onmessage');
+      console.error(e);
 #if ASSERTIONS
       var cmd = e['data']['cmd'];
       err('received "' + cmd + '" command from terminated worker: ' + worker.workerID);
 #endif
     };
+    worker.onerror = (e) => {
+      console.log('terminateWorker: worker.onerror');
+      console.error(e);
+    };
   },
 
   $killThread__deps: ['_emscripten_thread_free_data', '$terminateWorker'],
```

A minimal example:
main.c:
```c
void crash() {
  int (*fun)(void) = 0;
  (fun)();
}

int main() {
  crash();
  return 0;
}
```
Build:
```console
emcc -g -pthread -sPROXY_TO_PTHREAD -sDEFAULT_PTHREAD_STACK_SIZE=2MB -sSTACK_SIZE=5MB main.c -o main.html
```

But when I run `main.html`, the relevant console log is simply
<img width="1345" alt="image" src="https://user-images.githubusercontent.com/8726997/232325587-032bb2dc-e574-4151-b74d-6f80040eb8ce.png">

But if I have this snippet in `preamble.js`, even though the existing tests fail, somehow this simple program works as I intended. The console log in this case is:
<img width="1348" alt="image" src="https://user-images.githubusercontent.com/8726997/232325536-f9b2b053-dbda-4862-a200-1ae2581c05c8.png">
You can see the whole `ErrorEvent` is postmessaged to the main thread and printed, with its `error` field populated with the original error. And this is the behavior I want. Not sure how to make this happen in a correct way...